pkg/ingester/ingester.go

```diff
@@ -1001,68 +1051,93 @@ func (i *Ingester) startPushRequest(ctx context.Context, reqSize int64) (_ conte
 		inflightBytes = i.inflightPushRequestsBytes.Load()
 		rejectEqualInflightBytes = true // if inflightBytes == limit, reject new request
 	}
-
-	finishRequestInDefer := true
-
+	var instanceLimitsErr error
 	defer func() {
-		if finishRequestInDefer {
-			i.finishPushRequest(reqSize)
+		if instanceLimitsErr != nil {
```

Comment:
Since we now do all checks in single `checkInstanceLimits` function, there's no need for `defer` anymore, and we can move this logic after call to `checkInstanceLimits`.

Reply:
I have removed the `defer`.